SofaKernel/modules/SofaBaseMechanics/src/SofaBaseMechanics/DiagonalMass.inl

```diff
@@ -1152,11 +1140,142 @@ void DiagonalMass<DataTypes, MassType>::computeMass()
                 }
             }
             d_totalMass.setValue(total_mass);
-            d_vertexMass.endEdit();
         }
     }
 }
 
+
+template <class DataTypes, class MassType>
+typename DiagonalMass<DataTypes, MassType>::Real DiagonalMass<DataTypes, MassType>::computeVertexMass(const Real& md)
+{
+    Real total_mass = Real(0);
+
+    if (m_topology == nullptr)
+    {
+        msg_warning() << "No topology set. DiagonalMass can't computeMass.";
+        return total_mass;
+    }
+    
+    Real mass = Real(0);
+    helper::WriteAccessor<Data<MassVector> > masses = d_vertexMass;
+    // resize array
+    masses.clear();
+    masses.resize(this->mstate->getSize());
+    
+    // set to 0
+    for (unsigned int i = 0; i < masses.size(); ++i)
+        masses[i] = Real(0);
+
+    if (m_topology->getNbHexahedra() > 0 && hexaGeo)
+    {
+        m_massTopologyType = TopologyElementType::HEXAHEDRON;
+
+        for (Topology::HexahedronID i = 0; i < m_topology->getNbHexahedra(); ++i)
+        {
+            const Hexahedron& h = m_topology->getHexahedron(i);
+            if (hexaGeo)
```

Comment:
`hexaGeo` has already been checked for `nullptr` a few lines above

Reply:
no in fact "m_topology"  container pointer is checked, hexageo is the pointer to the geometry component.

Reply:
line 1169? `if (m_topology->getNbHexahedra() > 0 && hexaGeo)`

Reply:
ah no ok!